scrapy/utils/spider.py

```diff
@@ -4,17 +4,14 @@
 from scrapy.spiders import Spider
 from scrapy.utils.defer import deferred_from_coro
 from scrapy.utils.misc import arg_to_iter
-try:
-    from scrapy.utils.py36 import collect_asyncgen
-except SyntaxError:
-    collect_asyncgen = None
+from scrapy.utils.python import collect_asyncgen
 
 
 logger = logging.getLogger(__name__)
 
 
 def iterate_spider_output(result):
-    if collect_asyncgen and hasattr(inspect, 'isasyncgen') and inspect.isasyncgen(result):
```

Comment:
[`inspect.isasyncgen`](https://docs.python.org/3.6/library/inspect.html#inspect.isasyncgen) was added in Python 3.6